src/compiler/types.ts

```diff
@@ -3005,7 +3005,24 @@ namespace ts {
         isRestParameter?: boolean;
     }
 
-    export type SymbolTable = Map<Symbol>;
+    export type EscapedIdentifier = (string & { __escapedIdentifier: void }) | (void & { __escapedIdentifier: void });

```

Comment:
`UnderscoreEscapedString` is a bit verbose for such a common type. `EscapedName` might be better.